File: 1686-stone-game-vi/1686-stone-game-vi.cpp
class Solution {
public:
    int stoneGameVI(vector<int>& aliceValues, vector<int>& bobValues) {
        vector<vector<int>> sums;
        int n = aliceValues.size();
        
        for (int i = 0; i < n; ++i) {
            int a = aliceValues[i];
            int b = bobValues[i];
            sums.push_back({-(a + b), a, b});
        }
        
        sort(sums.begin(), sums.end());
        
        int alice = 0, bob = 0;
        
        for (int i = 0; i < n; ++i) {
            if (i % 2 == 0) {
                alice += sums[i][1];
            }
            else {
                bob += sums[i][2];
            }
        }
        
        if (alice > bob) {
            return 1;
        }
        else if (alice < bob) {
            return -1;
        }
        
        return 0;
    }
};

File: 1686-stone-game-vi/1686-stone-game-vi.py
class Solution:
    def stoneGameVI(self, aliceValues: List[int], bobValues: List[int]) -> int:
        tu_val = []
        n = len(aliceValues)
        
        for i in range(n):
            a = aliceValues[i]
            b = bobValues[i]
            tu_val.append((a + b, a, b))
            
        tu_val.sort()
        
        alice, bob = 0, 0
        
        for i in range(n):
            alice += tu_val[n - i - 1][1] if i % 2 == 0 else 0
            bob += tu_val[n - i - 1][2] if i % 2 == 1 else 0
            
        if alice > bob:
            return 1
        elif alice < bob:
            return -1
        
        return 0